### PR TITLE
Implementing v1

### PR DESCRIPTION
**Pull request by FlorianLoch**
🎉 

When agreeing on the general architecture I will add doc strings and a README :)

### CODE REVIEW

File: export.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
+package hibpsync
+
+import (
+	"fmt"
+	"io"
+)
+
+var lineSeparator = []byte("\n")
+
+func export(from, to int64, store storage, w io.Writer) error {
```
Comment (obitech) [inline export.go:10]: Can we add a comment to this method?

File: lib.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
+package hibpsync
```
Comment (obitech) [inline lib.go:1]: Again, let's add comments to everything exported here 😇

File: upstream.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+}
+
+type hibpResponse struct {
+	NotModified bool
+	ETag        string
+	Data        []byte
+}
+
+func (h *hibpClient) RequestRange(rangePrefix, etag string) (*hibpResponse, error) {
+	req, err := http.NewRequest("GET", h.endpoint+rangePrefix, nil)
```
Comment (obitech) [inline upstream.go:23]:
```suggestion
	req, err := http.NewRequest(http.MethodGet, h.endpoint+rangePrefix, nil)
```

**Review by obitech — COMMENTED** [review]
I know you said in the description you want to add docs/readme later but I'd rather do it now (I don't have gripes with the architecture), because this is public Github and not our internal git 😬

**Comment by FlorianLoch** [comment]
I did not necessarily suggest to not include documentation in this PR - I just didn't want to do it before getting a second opinion on the API 🙃 

But now that you took a look I will add some documentation.

**Comment by FlorianLoch** [comment]
Documentation has been added. PTAL :)

File: cmd/export/main.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
+// Small utility to export the HIBP data to stdout.
```
Comment (obitech) [inline cmd/export/main.go:1]:
```suggestion
// Package main contains a small utility to export the HIBP data to stdout.
```

File: cmd/sync/main.go [hunk 5]
```diff
@@ -0,0 +1,87 @@
+// Small utility to sync the HIBP data to the default data directory or to the directory specified as the
```
Comment (obitech) [inline cmd/sync/main.go:1]:
```suggestion
// Package main contains a small utility to sync the HIBP data to the default data directory or to the directory specified as the
```

**Review by obitech — COMMENTED** [review]
Very nice 🙌
Just two small comments to comply with the godoc best practices

**Review by obitech — COMMENTED** [review]
LGTM

**Review by obitech — APPROVED** [review]

**Pull request merged**